builder/azure/arm/step_create_resource_group.go

```diff
@@ -46,13 +48,17 @@ func (s *StepCreateResourceGroup) createResourceGroup(ctx context.Context, resou
 	return err
 }
 
-func (s *StepCreateResourceGroup) doesResourceGroupExist(ctx context.Context, resourceGroupName string) (bool, error) {
-	exists, err := s.client.GroupsClient.CheckExistence(ctx, resourceGroupName)
+func (s *StepCreateResourceGroup) doesResourceGroupExist(ctx context.Context, subscriptionId string, resourceGroupName string) (bool, error) {
+	id := commonids.NewResourceGroupID(subscriptionId, resourceGroupName)
+	exists, err := s.client.ResourceGroupsClient.Get(ctx, id)
 	if err != nil {
+		if exists.HttpResponse.StatusCode == 404 {
+			return false, nil
+		}
 		s.say(s.client.LastError.Error())
 	}
 
-	return exists.Response.StatusCode != 404, err
+	return exists.HttpResponse.StatusCode != 404, nil
```

Comment:
Does that happen? If the error might be linked to a `404`, should we do the check in both places?